src/ray/util/shared_lru.h

```diff
@@ -90,7 +96,8 @@ class SharedLruCache final {
   // Delete the entry with key `key`. Return true if the entry was found for
   // `key`, false if the entry was not found. In both cases, there is no entry
   // with key `key` existed after the call.
-  bool Delete(const Key &key) {
+  template <typename KeyLike>
+  bool Delete(KeyLike &&key) {
```

Comment:
why do we need a move in Delete? It's possible user wants to delete then still use its key.

Reply:
Also even if you want move semantics you want to pass in by value?

Reply:
It's perfect forward not move.

Reply:
can forward instead of pass by value in headers, slightly more efficient, but only makes sense for Put where you can possibly forward rvalue in, Get and Delete take const lvalue anyways so it doesn't matter...

Reply:
Put takes ownership of the key, so I take it by value;
Deletion and lookup only borrows reference, so perfect forwarding.